pyprocessing/math.py

```diff
@@ -162,191 +176,199 @@ def from_angle(angle, target=None):
         target.set(x, y)
         return target
 
-    def __init__(self, x=0, y=0, z=None):
-        self.x, self.y, self._z = x, y, z
-        self.is_2d = z is None
+    def __init__(self, *args, x=0, y=0, z=None):
+        if args:
+            self.seq = list(args)
+        else:
+            self.seq = [x, y, z] if z is not None else [x, y]
 
     def __copy__(self):
-        return self.copy()
+        return PVector(*self)
 
     def __add__(self, other):
-        if isinstance(other, (list, tuple)):
-            return self.add(*other)
-        elif isinstance(other, (PVector, complex)):
-            return self.add(other)
+        if isinstance(other, complex):
+            other = (other.real, other.imag)
+
+        if hasattr(other, '__iter__'):
+            return PVector(*(a + b for a, b in zip(self, other)))
+
         raise TypeError(
             f"Invalid operand '+' for type PVector and {type(other)}"
         )
 
     def __sub__(self, other):
-        if isinstance(other, (list, tuple)):
-            return self.sub(*other)
-        elif isinstance(other, (PVector, complex)):
-            return self.sub(other)
+        if isinstance(other, complex):
+            other = (other.real, other.imag)
+
+        if hasattr(other, '__iter__'):
+            return PVector(*(a - b for a, b in zip(self, other)))
+
         raise TypeError(
             f"Invalid operand '-' for type PVector and {type(other)}"
         )
 
     def __mul__(self, other):
-        if isinstance(other, PVector):
+        if isinstance(other, (float, int)):
+            return PVector(*(a * other for a in self))
```

Comment:
use a better variable name than 'a'

Reply:
I renamed it to elem_a and elem_b.
its hard to think of a good name for this variable.

Reply:
those are components of the vector

Reply:
Yeah, but I thought component_a + component_b is kinda long for a oneliner.

Reply:
you can shorten it to cmpnt if need be, although component is not that long, and you can add line break (while still following the style guide obviously)

Reply:
Ok going for line break then.